record/record_test.go

```diff
@@ -312,3 +313,137 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicReadRecovery(t *testing.T) {
+	buf := new(bytes.Buffer)
```

Comment:
It may be worth defining:

aaa := strings.Repeat("a", blockSize-headerSize)
bbb := strings.Repeat("b", blockSize-headerSize)
ccc := strings.Repeat("c", blockSize-headerSize)

at the top of this function.